dbt-snowflake/src/dbt/include/snowflake/macros/relations/table/create/iceberg_managed.sql

```diff
@@ -0,0 +1,57 @@
+{% macro snowflake__create_table_iceberg_managed_sql(relation, compiled_code) -%}
+{#-
+    Implements CREATE ICEBERG TABLE and CREATE ICEBERG TABLE ... AS SELECT (Snowflake as the Iceberg catalog):
+    https://docs.snowflake.com/en/sql-reference/sql/create-iceberg-table-snowflake
+
+    Limitations:
+    - Iceberg does not support temporary tables (use a standard Snowflake table)
+-#}
+
+{%- if not adapter.behavior.enable_iceberg_materializations.no_warn -%}
+    {%- do exceptions.raise_compiler_error('Was unable to create model as Iceberg Table Format. Please set the `enable_iceberg_materializations` behavior flag to True in your dbt_project.yml. For more information, go to https://docs.getdbt.com/reference/resource-configs/snowflake-configs#iceberg-table-format') -%}
+{%- endif -%}
+
+{%- set catalog_relation = adapter.build_catalog_relation(config.model) -%}
+
+{%- set copy_grants = config.get('copy_grants', default=false) -%}
+
+{%- set contract_config = config.get('contract') -%}
+{%- if contract_config.enforced -%}
+    {{- get_assert_columns_equivalent(compiled_code) -}}
+    {%- set compiled_code = get_select_subquery(compiled_code) -%}
+{%- endif -%}
```

Comment:
I'm a bit confused probably most about scoping more than anything else. Why is `compiled_code` set inside an `if` here then used below and what happens when it's empty?

Reply:
`compiled_code` is never empty if we got this far (or it shouldn't be at least). It gets passed in as an argument, and then if this model is contracted, we need to generate a "superquery" from the compiled code, as the subquery, that selects the specific columns from the compiled code. This will fail if the user has updated the model to have a different data schema. `get_select_subquery` updates the compiled code in place to wrap it with that select query.